server/controllers/BidController.js

```diff
@@ -47,7 +47,7 @@ const chooseBid = async (req, res) => {
   if (mission) {
     await addBidToCaptain(mission.vehicle_id, bidId);
     // droneApi.beginMission(mission.vehicle_id, mission.mission_id);
-    await updateVehicleStatus(mission.vehicle_id, 'contract_received');
+    await updateVehicleStatus(mission.vehicle_id, 'contract_received', mission.mission_id);
```

Comment:
the vehicle needs to pass the mission ID to the email sending function, because the mission ID has to be included in the URL inside the email - the URL which leads to the form

Reply:
do you mean for instance adding to BidController.js an additional call that will call gradd function?
from (BidController.js line 50):
    await updateVehicleStatus(mission.vehicle_id, 'contract_received', mission.mission_id);

to:
    await sendGraddEmail(mission.mission_id);
    await updateVehicleStatus(mission.vehicle_id, 'contract_received');

Reply:
Sounds good.